### PR TITLE
Web console: Fix the supervisor offset reset dialog.

### PR DESCRIPTION
**Pull request by vogievetsky**
fixes the dialog implemented here https://github.com/apache/druid/pull/14863

Basically the old dialog relied on `latestOffsets` to be set to work. This version does not. In addition it provides a fallback JSON view so it is easy to paste in the metadata.

image

Also allows starting a supervisor in a suspended state

### CODE REVIEW

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 1]
```diff
@@ -3246,6 +3246,27 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+              {
+                name: 'suspended',
+                type: 'boolean',
+                defined: isStreamingSpec,
+                defaultValue: false,
+                info: (
+                  <>
+                    <p>Submit this supervisor in a suspended state.</p>
+                    <p>
+                      Adding a suspended supervisor is useful if you are not ready to start
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3258]:
```suggestion
                      Submitting a suspended supervisor is helpful if you are not yet ready to begin ingesting data or if you prefer to configure the metadata for the
                      supervisor before starting it.
```

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 2]
```diff
@@ -3246,6 +3246,27 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
                 ),
               },
+              {
+                name: 'suspended',
+                type: 'boolean',
+                defined: isStreamingSpec,
+                defaultValue: false,
+                info: (
+                  <>
+                    <p>Submit this supervisor in a suspended state.</p>
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3256]:
```suggestion
                    <p>Create a supervisor in a suspended state.</p>
```

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 3]
```diff
@@ -3246,6 +3246,27 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+                    <p>
+                      Adding a suspended supervisor is useful if you are not ready to start
+                      inserting data yet or if you want to precisely configure the metadata for the
+                      supervisor before starting it.
+                    </p>
+                    <p>
+                      You can configure the exact offsets that the supervisor will read from from
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3263]:
```suggestion
You can configure the precise offsets that the supervisor will read from using the Actions menu on the Supervisor tab.
```

**Review by ektravel — COMMENTED** [review]
Left a few suggestions.

**Review by abhishekrb19 — APPROVED** [review]

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 4]
```diff
@@ -3246,6 +3246,28 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+                    <p>Create a supervisor in a suspended state.</p>
+                    <p>
+                      Creating a supervisor in a suspended state can be helpful if you are not yet
+                      ready to begin ingesting data or if you prefer to configure the supervisor's
+                      metadata before submitting it. inserting data yet or if you want to precisely
+                      configure the metadata for the supervisor before starting it.
+                    </p>
+                    <p>
+                      You can configure the exact offsets that the supervisor will read from using
+                      the Actions menu on the Supervisors tab. the action menu on the Supervisor tab
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3265]:
```suggestion
                      the Actions menu on the Supervisors tab.
```

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 5]
```diff
@@ -3246,6 +3246,28 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+                      after submitting a supervisor in a suspended state.
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3266]:
```suggestion

```

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 6]
```diff
@@ -3246,6 +3246,28 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+                type: 'boolean',
+                defined: isStreamingSpec,
+                defaultValue: false,
+                info: (
+                  <>
+                    <p>Create a supervisor in a suspended state.</p>
+                    <p>
+                      Creating a supervisor in a suspended state can be helpful if you are not yet
+                      ready to begin ingesting data or if you prefer to configure the supervisor's
+                      metadata before submitting it. inserting data yet or if you want to precisely
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3260]:
```suggestion
                      metadata before submitting it.
```

File: web-console/src/views/load-data-view/load-data-view.tsx [hunk 7]
```diff
@@ -3246,6 +3246,28 @@ export class LoadDataView extends React.PureComponent<LoadDataViewProps, LoadDat
...
+                      configure the metadata for the supervisor before starting it.
```
Comment (ektravel) [inline web-console/src/views/load-data-view/load-data-view.tsx:3261]:
```suggestion

```

**Review by ektravel — COMMENTED** [review]
@vogievetsky The original text didn't get removed after you committed my suggestions.

**Comment by vogievetsky** [comment]
Ok got it!

**Pull request merged**